New translate features

Birdulon opened the pull request:
## Description
All arguments to `translate` will now be converted to strings. If a `TextStrings` is passed (i.e. from the textmap lookups), it will pick the correct localised string from within.

Cleans up commands that were using `Integer::toString` and `Float::toString` for `translate` arguments.

Adds talent name and desc feedback to the talent command.
![image](https://user-images.githubusercontent.com/5397662/183925212-99a5fdc6-4c07-44ad-bb60-35c65177a598.png)
![image](https://user-images.githubusercontent.com/5397662/183925252-c0e4044c-c6d0-4fe2-a60c-0c5e47cf8b96.png)

Will probably make some other tweaks to the language backend. Might even get around to handbook search.


## Issues fixed by this PR
## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

KingRainbow44 commented:
a component system (similar to how [Paper](https://papermc.io) does it) might be nice with grasscutter (only maybe, would be hell to convert everything to it) as colors in-game and in the server console are represented differently, though this should probably be another PR.

Birdulon commented:
We can probably settle for stripping xml tags when printing to console

Birdulon committed:
Improve translate function
Make commands pass ints and floats instead of manually casting them to string first

Birdulon committed:
Make TalentCommand show off new Translate function
